Sources/Core/AWSSDKIdentity/Sources/AWSSDKIdentity/AWSCredentialIdentityResolvers/ProfileAWSCredentialIdentityResolver.swift

```diff
@@ -191,8 +208,16 @@ public struct ProfileAWSCredentialIdentityResolver: AWSCredentialIdentityResolve
                     source: credSource,
                     identityProperties: identityProperties
                 )
+                let sourceCredsWithUpdatedFeatureIDs = prependFeatureIDs(
+                    featureIDsToPrepend: credentialFeatureIDs + [
+                        CredentialFeatureID.CREDENTIALS_PROFILE_NAMED_PROVIDER.rawValue
+                    ],
+                    creds: sourceCreds
+                )
```

Comment:
In the case of `credential_source`, prepend `"p"` to the feature ID list. This is because the credential source providers (IMDS, ECS, IMDS (EC2)) only return feature IDs relevant to them via returned credentials' properties. Profile chain is the entity that knows the code path was via `credential_source` delegation.